src/connectors/articleService.ts

```diff
@@ -1385,27 +1385,33 @@ export class ArticleService extends BaseService {
     }
 
     // logged-in user
-    // calculate heart beat lapsed time in secondes
+    // calculate heart beat lapsed time in milisecondes
     const lapse = Date.now() - new Date(oldData.updatedAt).getTime()
 
     // calculate last read total time
     const readLength = Date.now() - new Date(oldData.lastRead).getTime()
 
-    // if original read longer than 30 minutes
-    // skip
-    if (userId && readLength > MINUTE * 30) {
+    // if lapse shorter than 5 minutes or readLength longer than 30 minutes, 
+    // only update the updatedAt and then return
+   if (lapse < MINUTE * 5 || readLength > MINUTE * 30) {
```

Comment:
Should be `lapse < MINUTE * 5 && readLength > MINUTE * 30`, since we want to stop recording when the accumulated time is larger than 30 minutes. If we stop recording when lapse is smaller than 5 minute **or** when  accumulated time is larger than 30 minutes, then it will never record since normally there's less than 5 minutes lapse between heartbeats (frontend calls every 5 seconds).